components/sync/driver/brave_sync_service_impl.cc

```diff
@@ -175,4 +182,113 @@ SyncServiceCrypto* BraveSyncServiceImpl::GetCryptoForTests() {
   return &crypto_;
 }
 
+namespace {
+const int kMaxPermanentlyDeleteSyncAccountAttempts = 5;
+const int kDelayBetweenDeleteSyncAccountAttemptsMsec = 500;
+}  // namespace
+
+void BraveSyncServiceImpl::OnAccountDeleted(
+    const int current_attempt,
+    base::OnceCallback<void(const SyncProtocolError&)> callback,
+    const SyncProtocolError& sync_protocol_error) {
+  if (sync_protocol_error.error_type == SYNC_SUCCESS) {
+    std::move(callback).Run(sync_protocol_error);
+    // If request succeded - reset and clear all in a forced way
+    // The code below cleans all on an initiator device. Other devices in the
+    // chain will be cleaned at BraveSyncServiceImpl::ResetEngine
+    DCHECK(initiated_delete_account_);
+    BraveSyncServiceImpl::StopAndClear();
+  } else if (current_attempt < kMaxPermanentlyDeleteSyncAccountAttempts) {
+    // Server responded failure, but we need to try more
+    base::SequencedTaskRunnerHandle::Get()->PostDelayedTask(
+        FROM_HERE,
+        base::BindOnce(&BraveSyncServiceImpl::PermanentlyDeleteAccountImpl,
+                       weak_ptr_factory_.GetWeakPtr(), current_attempt + 1,
+                       std::move(callback)),
+        base::Milliseconds(kDelayBetweenDeleteSyncAccountAttemptsMsec));
+  } else {
+    // Server responded failure, and we are out of our attempts
+    initiated_delete_account_ = false;
+    std::move(callback).Run(sync_protocol_error);
+  }
+}
+
+void BraveSyncServiceImpl::PermanentlyDeleteAccountImpl(
+    const int current_attempt,
+    base::OnceCallback<void(const SyncProtocolError&)> callback) {
+  if (!engine_) {
+    // We can reach here if two devices almost at the same time will initiate
+    // the deletion procedure
+    SyncProtocolError sync_protocol_error;
+    sync_protocol_error.error_type = SYNC_SUCCESS;
+    std::move(callback).Run(sync_protocol_error);
+    return;
+  }
+
+  DCHECK_GE(current_attempt, 1);
+  DCHECK_NE(current_attempt, 10);
+
+  engine_->PermanentlyDeleteAccount(base::BindOnce(
+      &BraveSyncServiceImpl::OnAccountDeleted, weak_ptr_factory_.GetWeakPtr(),
+      current_attempt, std::move(callback)));
+}
+
+void BraveSyncServiceImpl::PermanentlyDeleteAccount(
+    base::OnceCallback<void(const SyncProtocolError&)> callback) {
+  initiated_delete_account_ = true;
+  PermanentlyDeleteAccountImpl(1, std::move(callback));
+}
+
+void BraveSyncServiceImpl::ResetEngine(ShutdownReason shutdown_reason,
+                                       ResetEngineReason reset_reason) {
+  SyncServiceImpl::ResetEngine(shutdown_reason, reset_reason);
+
+  if (initiated_self_device_info_deleted_) {
+    return;
+  }
+
+  if (shutdown_reason == ShutdownReason::DISABLE_SYNC_AND_CLEAR_DATA &&
+      reset_reason == ResetEngineReason::kDisabledAccount &&
+      sync_disabled_by_admin_ && !initiated_delete_account_ &&
```

Comment:
where is the server code that will send `DISABLED_BY_ADMIN` when existing client trying to connect to deleted sync chain?

Reply:
It is here
https://github.com/brave/go-sync/compare/master...102-add-account-delete#diff-3b16b902a214eca6845aed1de9a23dfabedb90f6f7e234c928e574b0cd230a1cR37